site/sfguides/src/native-app-chairlift/native-app-chairlift.md

```diff
@@ -0,0 +1,632 @@
+
+```plaintext
+src/
+├─ README.md
+├─ manifest.yml
+├─ setup.sql
+├─ ui/
+│  ├─ chairlift_data.py
+│  ├─ environment.yml
+│  ├─ first_time_setup.py
+│  ├─ references.py
+│  ├─ ui_common.py
+│  ├─ util.py
+│  ├─ v_configuration.py
+│  ├─ v_dashboard.py
+│  ├─ v_sensor_data.py
+```
+
+This directory contains the source code for the native app. This Quickstart uses **src/** as the name of the folder, mostly for the purposes of convention. In practice, this folder may take on any name you'd like. Here's an overview of what this folder contains:
+
+**manifest.yml** 
+
+- A manifest file is a requirement when creating a native app. This file defines the configuration and setup properties required by the application. It contains metadata about the app (version, etc.), artifacts required by the app, and log configuration settings. It also defines the privileges that the consumer must grant when the application is installed and run in their account. Finally, it also contains references defined by the provider. Typically these references refer to tables and the corresponding privileges needed by the app to run against consumer data. For more information, see [Creating the Manifest File](https://docs.snowflake.com/en/developer-guide/native-apps/creating-manifest).
+
+**setup.sql** 
+
+- A setup script is a requirement when creating a native app. This script contains statements that are run when the consumer installs or upgrades an application, or when a provider installs or upgrades an application for testing. The location of this script should be specified in the manifest file. For more information, see [Creating the Setup Script](https://docs.snowflake.com/en/developer-guide/native-apps/creating-setup-script).
+
+**README.md** 
+
+- This file should provide a description of what the app does. This file is shown when viewing the app within Snowflake.
+
+**ui/** 
+
+- This directory is specific to this Quickstart, and contains all of the files and code used to create the front-end of the app. Front-ends for Snowflake Native Apps are built with Streamlit. You should peruse all of the files in this folder to get familiar with how the front-end is built, and pay special attention to the files that define the main views within the app, namely **v_dashboard.py**, **v_sensor_data.py**, and **v_configuration.py**. For more information, see [Adding Frontend Experience to Your Application with Streamlit](https://docs.snowflake.com/en/developer-guide/native-apps/adding-streamlit).
+
+
+<!-- ------------------------ -->
+## Set up account roles
+Duration: 3
+
+Let's start building the app. You'll first need to configure certain roles and permissions within your Snowflake account. This will allow you to view the app as an app admin (perhaps the chairlift manufacturer is doing some diagnostic work), or as an app viewer (perhaps the ski resort owner is interested in identifying chairlifts in need of repair).
```

Comment:
Both roles exist solely on the consumer side, so the example about `app_admin` is incorrect. How about:

- `app_admin`: used for configuring the application after it is installed and dismissing warnings

Reply:
App admin sets up the account, app viewer – interested parties.